### PR TITLE
chore: Create commonly used widgets

### PR DESCRIPTION
**Pull request by Neha-Rode**
# Description of Changes
Created a submit button, organization tile, featured bulletin tile, community bulletin tile, tag list, and section banner. Each of these is highly customizable and is intended for use by others creating screens for concord-mobile. 

# Other Notes
The `TagPicker` widget can be used when the user has the ability to add and remove tags from the list and also when it is a static list of tags that the user is not meant to interact with. This only requires providing some extra arguments to the `TagPicker` instance (the default is that they can remove and add tags).

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -20,7 +22,7 @@ class Concord extends StatelessWidget {
         title: 'Concord',
         // For testing, change the assignment of home to be whatever screen you're working on
         // Example: home: AuthScreen()
-        home: HomeScreen(),
+        home: RequestDetails(),
```
Comment (WillieCubed) [inline lib/main.dart:25]:
```suggestion
        home: HomeScreen(),
```

I'd revert this change because it's not directly related to the changes of this PR.

File: lib/ui/screens/request_details_screen.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import 'package:concord/ui/widgets/concord_tiles.dart';
+import 'package:concord/ui/widgets/layout_widgets.dart';
+import 'package:concord/ui/widgets/submit_button.dart';
+import 'package:concord/ui/widgets/tags.dart';
+import 'package:flutter/material.dart';
+import '../../utils/locator.dart';
+import '../theme/theme.dart';
+
```
Comment (WillieCubed) [inline lib/ui/screens/request_details_screen.dart:8]: Add a doc comment describing what this component is for.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (WillieCubed) [inline .vscode/launch.json:1]: Not everyone on the team uses VS Code, so this file shouldn't be checked into source control. To fix this:
- Delete the file on your local machine
- Commit the change
- Update `.gitignore` to exclude `.vscode`
- Commit that change
- Undo the file deletion

Comment (WillieCubed) [reply .vscode/launch.json:1]: Also, I'd recommend adding this in the README under the "Getting Started" section in a part labeled "Using VS Code".

File: lib/ui/widgets/concord_tiles.dart [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+class CommunityBulletinTile extends StatelessWidget {
+  final Asset image;
+  final String title;
+  final String description;
+  final Color titleColor;
+  final Color descriptionColor;
+
+  const CommunityBulletinTile({
+    Key key,
+    // TODO:  set placeholder image if one not provided
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:13]: Refrain from merging TODOs into `dev`. If needed, create an issue to better document what needs to be done.
```suggestion
```

File: lib/ui/widgets/concord_tiles.dart [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+            ),
+            _textPortion()
+          ],
+        ) : 
+        //if there is no image
+        _textPortion()
+    );
+  }
+
+  Padding _textPortion(){
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:59]:
```suggestion
  Padding _textPortion() {
```

File: lib/ui/widgets/concord_tiles.dart [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    this.descriptionColor
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    double radius = 4.0;
+    double height = 98;
+
+    return Container(
+      //if there is an image, the tile will be longer to accommodate the thumb nail image
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:27]: Style nit: Put spaces between comment delimiters and content and capitalize starts of sentences
```suggestion
      // If there is an image, the tile will be longer to accommodate the thumbnail image
```

File: lib/ui/widgets/concord_tiles.dart [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+          BoxShadow(
+            color: Colors.grey.withOpacity(0.6),
+            spreadRadius: 5,
+            blurRadius: 7,
+            offset: Offset(0,3)
+          )
+        ]
+      ),
+      child: image != null ?
+        //if there is an image
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:43]: This comment can already be explained through the ternary statement you provide `image != null`, so it can be safely removed.

```suggestion
```

File: lib/ui/widgets/concord_tiles.dart [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+          children: <Widget>[
+            AssetThumb(
+              asset: image,
+              width: height.floor(),
+              height: height.floor(),
+            ),
+            _textPortion()
+          ],
+        ) : 
+        //if there is no image
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:54]: See the change to line 43.

```suggestion
        //if there is no image
```

File: lib/ui/widgets/concord_tiles.dart [hunk 9]
```diff
@@ -0,0 +1,245 @@
+import 'package:flutter/material.dart';
+import 'package:multi_image_picker/multi_image_picker.dart';
+
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:3]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/concord_tiles.dart [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+      title,
+      style: TextStyle(
+        color: titleColor ?? Colors.black,
+        fontSize: 24,
+        fontWeight: FontWeight.w500
+      )
+    );
+  }
+}
+
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:103]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/concord_tiles.dart [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+                ],
+              ),
+            ),
+          )
+        ],
+      )
+    );
+  }
+}
+
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:181]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/layout_widgets.dart [hunk 12]
```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/material.dart';
+
+import '../../utils/locator.dart';
+import '../theme/theme.dart';
+
```
Comment (WillieCubed) [inline lib/ui/widgets/layout_widgets.dart:5]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/submit_button.dart [hunk 13]
```diff
@@ -0,0 +1,57 @@
+import 'package:flutter/material.dart';
+import '../../utils/locator.dart';
+import '../theme/theme.dart';
+
```
Comment (WillieCubed) [inline lib/ui/widgets/submit_button.dart:4]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/tags.dart [hunk 14]
```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_tags/flutter_tags.dart';
+import '../../utils/locator.dart';
+import '../theme/theme.dart';
+
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:5]: Add a doc comment here describing the purpose of this widget.

File: lib/ui/widgets/tags.dart [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+  final Color chipColor;
+  final Color borderColor;
+  final Color removeButtonColor;
+  final Color textColor;
+  final List<String> initTags;
+  final List<String> suggestions;
+  final bool horizontalScrollEnabled;
+  final String hintText;
+  List<String> tags;
+
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:18]: Add a doc comment here explaining each of the parameters for this constructor.

File: lib/ui/widgets/tags.dart [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+    this.removeButtonColor,
+    this.hintText = "Add a skill or certification",
+    this.horizontalScrollEnabled = false
+  }) : super(key: key);
+
+  
+  @override
+  _TagPickerState createState() => _TagPickerState();
+}
+
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:38]: Add a doc comment here describing what this state keeps track of.

File: lib/ui/widgets/tags.dart [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+        suggestionTextColor: _appTheme.secondaryText,
+        suggestions: widget.suggestions ?? null,
+        onSubmitted: (String string){
+          setState(() {
+            widget.tags.add(string);
+          });
+        }
+      ),
+      itemCount: widget.tags != null ? widget.tags.length : 0,
+      itemBuilder: (index){
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:73]:
```suggestion
      itemBuilder: (index) {
```

File: lib/ui/widgets/tags.dart [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+          ),
+          elevation: 0.0,
+          activeColor: widget.chipColor ?? _appTheme.bannerColor,
+          borderRadius: BorderRadius.circular(30),
+          combine: ItemTagsCombine.onlyText,
+          textActiveColor: widget.textColor ?? _appTheme.secondaryText,
+          removeButton: widget.removable ?
+            ItemTagsRemoveButton(
+              backgroundColor: widget.removeButtonColor ?? _appTheme.secondaryText,
+              onRemoved: (){
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:92]:
```suggestion
              onRemoved: () {
```

File: lib/ui/widgets/tags.dart [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+  @override
+  _TagPickerState createState() => _TagPickerState();
+}
+
+class _TagPickerState extends State<TagPicker> {
+
+  final ConcordThemeData _appTheme = locator<ConcordThemeManager>().theme;
+
+  @override
+  void initState(){
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:44]:
```suggestion
  void initState() {
```

File: lib/ui/widgets/concord_tiles.dart [hunk 20]
```diff
@@ -0,0 +1,245 @@
...
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            _title()
+          ],
+        ),
+    );
+  }
+
+  Text _title(){
```
Comment (WillieCubed) [inline lib/ui/widgets/concord_tiles.dart:92]:
```suggestion
  Text _title() {
```

File: lib/ui/widgets/tags.dart [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+      horizontalScroll: widget.horizontalScrollEnabled,
+      runSpacing: 8,
+      textField: TagsTextField(
+        enabled: widget.textFieldEnabled,
+        autofocus: false,
+        hintText: widget.hintText,
+        hintTextColor: Colors.grey,
+        suggestionTextColor: _appTheme.secondaryText,
+        suggestions: widget.suggestions ?? null,
+        onSubmitted: (String string){
```
Comment (WillieCubed) [inline lib/ui/widgets/tags.dart:66]:
```suggestion
        onSubmitted: (String string) {
```

File: lib/ui/widgets/layout_widgets.dart [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+    @required this.title,
+    this.bannerColor,
+    this.titleColor,
+    this.fontSize
+  }) : super(key: key);
+  
+  @override
+  Widget build(BuildContext context) {
+
+    ConcordThemeData _appTheme = locator<ConcordThemeManager>().theme;
```
Comment (WillieCubed) [inline lib/ui/widgets/layout_widgets.dart:23]: Shouldn't this be `final`?
```suggestion
   final ConcordThemeData _appTheme = locator<ConcordThemeManager>().theme;
```

File: .vscode/launch.json (hunk 3 above)
Comment (Neha-Rode) [reply .vscode/launch.json]: Alright, I'll do that
